edk2toollib/uefi/edk2/parsers/base_parser.py

```diff
@@ -71,37 +81,35 @@ def SetInputVars(self, inputdict):
 
     def FindPath(self, *p):
         """
-
+        Given a path, it will find it relative to the root, the current target file, or the packages path
         Args:
           *p:
 
-        Returns:
+        Returns: None on failure
 
         """
-        # NOTE: Some of this logic should be replaced
-        #       with the path resolution from Edk2Module code.
-
         # If the absolute path exists, return it.
```

Comment:
you never do this?  You never treat it as absolute.  If that is supported you should check it and then not do any path modifications if it is absolute. Also seems unit test should have caught this if it was intended to work

Reply:
This doesn't seem to make any sense?

Reply:
I would have expected the code to if os.path.isabs(p) and os.path.exists(p) 

The code never checks if it is absolute.  It always joins it with rootpath which would not work if the incoming path was something like c:\MyFolder\MyFile.dsc  

This would turn out to be something c:\MyFolder\c:\MyFolder\MyFile.dsc

Does that make sense?

Reply:
So the code you have highlighted isn't clear what you're commenting on